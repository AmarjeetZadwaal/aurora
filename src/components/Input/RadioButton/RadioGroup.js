import React from "react";
import PropTypes from "prop-types";
import Provider from "../../SelectionProvider/Provider";
import KeyBoardProvider from "../../KeyboardNavigation/Provider";

const RadioGroup = ({
  onChange,
  children,
  value,
  ariaLabel,
  ariaLabelledBy
}) => (
  <Provider onChange={onChange} isMultiSelect={false} value={value}>
    <KeyBoardProvider
      role="radiogroup"
      aria-label={ariaLabel}
      aria-labelledby={ariaLabelledBy}
    >
      {children}
    </KeyBoardProvider>
  </Provider>
);

RadioGroup.propTypes = {
  onChange: PropTypes.func,
  children: PropTypes.node,
  value: PropTypes.arrayOf(PropTypes.string),
  ariaLabel: PropTypes.string,
  ariaLabelledBy: PropTypes.string
};

RadioGroup.defaultProps = {
  children: null,
  onChange: null,
  value: [],
  ariaLabel: "",
  ariaLabelledBy: ""
};

RadioGroup.displayName = "RadioGroup";

export default RadioGroup;
