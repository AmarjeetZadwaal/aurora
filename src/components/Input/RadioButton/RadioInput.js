import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import colors from "../../../theme/colors";

const RadioInput = styled.input.attrs({
  type: "radio"
})`
  margin: 0;
  appearance: none;
  position: relative;
  outline: none;
  .radio-button--large & {
    width: 24px;
    height: 24px;
  }
  .radio-button--small & {
    width: 16px;
    height: 16px;
  }
  &:before {
    content: "";
    position: absolute;
    background-color: transparent;
    border-radius: 50%;
    border: 1px solid ${colors.onyx.light};

    .radio-button--large & {
      width: 24px;
      height: 24px;
    }
    .radio-button--small & {
      width: 16px;
      height: 16px;
    }
    &.radio-button--disabled {
      color: grey;
    }
  }
  &:after {
    content: "";
    opacity: 0;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    border-radius: 50%;
    background-color: ${colors.azure.base};

    .radio-button--large & {
      width: 12px;
      height: 12px;
    }
    .radio-button--small & {
      width: 8px;
      height: 8px;
    }
  }
  &:hover:before {
    border-width: 2px;
    border-color: ${colors.azure.base};
  }
  &:focus:before {
    border-width: 2px;
    border-color: ${colors.azure.base};
  }
  &:checked:after {
    opacity: 1;
    position: absolute;
  }
`;

class RadioInputComponent extends React.Component {
  static propTypes = {
    size: PropTypes.oneOf(["large", "small"]),
    isActive: PropTypes.bool,
    name: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
    checked: PropTypes.bool.isRequired,
    onClick: PropTypes.func,
    isFocused: PropTypes.bool.isRequired
  };

  static defaultProps = {
    size: "small",
    isActive: true,
    onClick: null
  };

  constructor(props) {
    super(props);
    this.SelectedElement = React.createRef();
  }

  componentDidUpdate() {
    this.props.isFocused && this.SelectedElement.current.focus(); // eslint-disable-line
  }

  render() {
    const { name, value, size, isActive, checked, ...props } = this.props;

    return (
      <RadioInput
        value={value}
        name={name}
        size={size}
        id={`${name + value}input`}
        disabled={!isActive}
        checked={checked}
        aria-labelledby={`${name + value}label`}
        aria-checked={checked}
        role="radio"
        {...props}
        innerRef={this.SelectedElement}
      />
    );
  }
}
export default RadioInputComponent;
