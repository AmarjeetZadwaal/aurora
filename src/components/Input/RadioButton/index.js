/* eslint import/prefer-default-export: off */
export { default as RadioGroup } from "./RadioGroup";
