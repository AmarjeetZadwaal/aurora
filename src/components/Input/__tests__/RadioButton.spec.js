import React from "react";
import renderer from "react-test-renderer";
import { renderIntoDocument, cleanup, fireEvent } from "react-testing-library";
import RadioGroup from "../RadioButton/RadioGroup";
import RadioButton from "../RadioButton/RadioButton";

describe("RadioGroup", () => {
  afterEach(cleanup);
  afterAll(cleanup);

  it("renders default input", () => {
    expect(renderGroupComponent()).toMatchSnapshot();
  });

  it("renders small input", () => {
    expect(renderGroupComponent({ size: "small" })).toMatchSnapshot();
  });

  it("renders large input", () => {
    expect(renderGroupComponent({ size: "large" })).toMatchSnapshot();
  });

  it("onChange recives current value", () => {
    const onChange = jest.fn();
    const { getByTestId } = renderIntoDocument(
      <RadioGroup onChange={onChange}>
        <RadioButton
          name="something"
          data-testid="test-radiobutton"
          value="else"
          size="small"
          index={0}
        />
      </RadioGroup>
    );
    fireEvent.change(getByTestId("test-radiobutton"));
    expect(onChange).toHaveBeenCalledWith(["else"]);
  });

  it("only one item selected", () => {
    const onChange = jest.fn();
    const { getByTestId } = renderIntoDocument(
      <RadioGroup onChange={onChange}>
        <RadioButton
          name="something"
          data-testid="test-radiobutton"
          value="else"
          size="small"
          index={0}
        />
        <RadioButton
          name="something"
          data-testid="test-radiobutton2"
          value="new"
          size="small"
          index={1}
        />
      </RadioGroup>
    );
    fireEvent.change(getByTestId("test-radiobutton"));
    fireEvent.change(getByTestId("test-radiobutton2"));
    expect(onChange).toHaveBeenCalledWith(["new"]);
  });

  it("passing in a slected item", () => {
    const onChange = jest.fn();
    const { getByTestId } = renderIntoDocument(
      <RadioGroup value={["value1"]} onChange={onChange}>
        <RadioButton
          name="name1"
          data-testid="test-radiobutton"
          value="value1"
          size="small"
          index={0}
        />
        <RadioButton
          name="name2"
          data-testid="test-radiobutton2"
          value="value2"
          size="small"
          index={1}
        />
        <RadioButton
          name="name3"
          data-testid="test-radiobutton3"
          value="value3"
          size="small"
          index={2}
        />
      </RadioGroup>
    );

    expect(getByTestId("test-radiobutton").hasAttribute("checked"));
    expect(!getByTestId("test-radiobutton2").hasAttribute("checked"));
    expect(!getByTestId("test-radiobutton3").hasAttribute("checked"));
  });

  it("passing in a slected item then selecting another", () => {
    const onChange = jest.fn();
    const { getByTestId } = renderIntoDocument(
      <RadioGroup value={["value1"]} onChange={onChange}>
        <RadioButton
          name="name1"
          data-testid="test-radiobutton"
          value="value1"
          size="small"
          index={0}
        />
        <RadioButton
          name="name2"
          data-testid="test-radiobutton2"
          value="value2"
          size="small"
          index={1}
        />
        <RadioButton
          name="name3"
          data-testid="test-radiobutton3"
          value="value3"
          size="small"
          index={2}
        />
      </RadioGroup>
    );

    fireEvent.change(getByTestId("test-radiobutton2"));

    expect(!getByTestId("test-radiobutton").hasAttribute("checked"));
    expect(getByTestId("test-radiobutton2").hasAttribute("checked"));
    expect(!getByTestId("test-radiobutton3").hasAttribute("checked"));
  });

  function renderGroupComponent(props = {}) {
    return renderer.create(
      <RadioGroup {...props}>
        {" "}
        <RadioButton name="name1" value="value" size="small" index={0} />{" "}
      </RadioGroup>
    );
  }
});
