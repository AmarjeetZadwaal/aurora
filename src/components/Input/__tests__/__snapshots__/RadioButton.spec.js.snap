// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup renders default input 1`] = `
.c1 {
  margin: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  position: relative;
  outline: none;
}

.radio-button--large .c1 {
  width: 24px;
  height: 24px;
}

.radio-button--small .c1 {
  width: 16px;
  height: 16px;
}

.c1:before {
  content: "";
  position: absolute;
  background-color: transparent;
  border-radius: 50%;
  border: 1px solid rgba(0,0,15,0.6);
}

.radio-button--large .c1:before {
  width: 24px;
  height: 24px;
}

.radio-button--small .c1:before {
  width: 16px;
  height: 16px;
}

.c1:before.radio-button--disabled {
  color: grey;
}

.c1:after {
  content: "";
  opacity: 0;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  border-radius: 50%;
  background-color: rgba(2,108,223,1);
}

.radio-button--large .c1:after {
  width: 12px;
  height: 12px;
}

.radio-button--small .c1:after {
  width: 8px;
  height: 8px;
}

.c1:hover:before {
  border-width: 2px;
  border-color: rgba(2,108,223,1);
}

.c1:focus:before {
  border-width: 2px;
  border-color: rgba(2,108,223,1);
}

.c1:checked:after {
  opacity: 1;
  position: absolute;
}

.c0 {
  position: relative;
  cursor: pointer;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: left;
  color: rgba(0,0,15,1);
  outline: none;
}

.c0.radio-button--disabled {
  color: grey;
  cursor: default;
  pointer-events: none;
}

.c0.radio-button--large {
  font-size: 16px;
  line-height: 1.5;
}

.c0.radio-button--small {
  font-size: 14px;
  line-height: 1.5;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  outline: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 7px;
}

<div
  onKeyDown={[Function]}
  role="radiogroup"
>
   
  <label
    className="radio-button--small c0"
    id="name1valuelabel"
  >
    <input
      aria-checked={false}
      aria-labelledby="name1valuelabel"
      checked={false}
      className="c1"
      disabled={false}
      id="name1valueinput"
      name="name1"
      onChange={[Function]}
      onClick={null}
      role="radio"
      size="small"
      type="radio"
      value="value"
    />
    <span
      className="c2"
    />
  </label>
   
</div>
`;

exports[`RadioGroup renders large input 1`] = `
.c1 {
  margin: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  position: relative;
  outline: none;
}

.radio-button--large .c1 {
  width: 24px;
  height: 24px;
}

.radio-button--small .c1 {
  width: 16px;
  height: 16px;
}

.c1:before {
  content: "";
  position: absolute;
  background-color: transparent;
  border-radius: 50%;
  border: 1px solid rgba(0,0,15,0.6);
}

.radio-button--large .c1:before {
  width: 24px;
  height: 24px;
}

.radio-button--small .c1:before {
  width: 16px;
  height: 16px;
}

.c1:before.radio-button--disabled {
  color: grey;
}

.c1:after {
  content: "";
  opacity: 0;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  border-radius: 50%;
  background-color: rgba(2,108,223,1);
}

.radio-button--large .c1:after {
  width: 12px;
  height: 12px;
}

.radio-button--small .c1:after {
  width: 8px;
  height: 8px;
}

.c1:hover:before {
  border-width: 2px;
  border-color: rgba(2,108,223,1);
}

.c1:focus:before {
  border-width: 2px;
  border-color: rgba(2,108,223,1);
}

.c1:checked:after {
  opacity: 1;
  position: absolute;
}

.c0 {
  position: relative;
  cursor: pointer;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: left;
  color: rgba(0,0,15,1);
  outline: none;
}

.c0.radio-button--disabled {
  color: grey;
  cursor: default;
  pointer-events: none;
}

.c0.radio-button--large {
  font-size: 16px;
  line-height: 1.5;
}

.c0.radio-button--small {
  font-size: 14px;
  line-height: 1.5;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  outline: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 7px;
}

<div
  onKeyDown={[Function]}
  role="radiogroup"
>
   
  <label
    className="radio-button--small c0"
    id="name1valuelabel"
  >
    <input
      aria-checked={false}
      aria-labelledby="name1valuelabel"
      checked={false}
      className="c1"
      disabled={false}
      id="name1valueinput"
      name="name1"
      onChange={[Function]}
      onClick={null}
      role="radio"
      size="small"
      type="radio"
      value="value"
    />
    <span
      className="c2"
    />
  </label>
   
</div>
`;

exports[`RadioGroup renders small input 1`] = `
.c1 {
  margin: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  position: relative;
  outline: none;
}

.radio-button--large .c1 {
  width: 24px;
  height: 24px;
}

.radio-button--small .c1 {
  width: 16px;
  height: 16px;
}

.c1:before {
  content: "";
  position: absolute;
  background-color: transparent;
  border-radius: 50%;
  border: 1px solid rgba(0,0,15,0.6);
}

.radio-button--large .c1:before {
  width: 24px;
  height: 24px;
}

.radio-button--small .c1:before {
  width: 16px;
  height: 16px;
}

.c1:before.radio-button--disabled {
  color: grey;
}

.c1:after {
  content: "";
  opacity: 0;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  border-radius: 50%;
  background-color: rgba(2,108,223,1);
}

.radio-button--large .c1:after {
  width: 12px;
  height: 12px;
}

.radio-button--small .c1:after {
  width: 8px;
  height: 8px;
}

.c1:hover:before {
  border-width: 2px;
  border-color: rgba(2,108,223,1);
}

.c1:focus:before {
  border-width: 2px;
  border-color: rgba(2,108,223,1);
}

.c1:checked:after {
  opacity: 1;
  position: absolute;
}

.c0 {
  position: relative;
  cursor: pointer;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: left;
  color: rgba(0,0,15,1);
  outline: none;
}

.c0.radio-button--disabled {
  color: grey;
  cursor: default;
  pointer-events: none;
}

.c0.radio-button--large {
  font-size: 16px;
  line-height: 1.5;
}

.c0.radio-button--small {
  font-size: 14px;
  line-height: 1.5;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  outline: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 7px;
}

<div
  onKeyDown={[Function]}
  role="radiogroup"
>
   
  <label
    className="radio-button--small c0"
    id="name1valuelabel"
  >
    <input
      aria-checked={false}
      aria-labelledby="name1valuelabel"
      checked={false}
      className="c1"
      disabled={false}
      id="name1valueinput"
      name="name1"
      onChange={[Function]}
      onClick={null}
      role="radio"
      size="small"
      type="radio"
      value="value"
    />
    <span
      className="c2"
    />
  </label>
   
</div>
`;
